### PR TITLE
Background images

### PR DESCRIPTION
**Pull request by anttileppa**
Adds basic support for using background images

### CODE REVIEW

**Comment by daohoangson** [comment]
This is great!

I wanted to give you a quick heads-up that I'm currently in the process of getting v0.14 ready for release, and it's been quite a journey so far (take a look at #897). It might take a little while to review this PR and #1021. After we merge the new version, there might be some conflicts, but no worries! The underlying architecture is still quite similar, and I'm here to lend a hand in ironing out any kinks. Thank you for your contributions!

File: packages/core/lib/src/core_widget_factory.dart [hunk 1]
```diff
@@ -188,6 +193,31 @@ class WidgetFactory {
     }
   }
 
+  /// Builds decoration image from [url]
+  DecorationImage? _buildDecorationImage(String? url) {
```
Comment (daohoangson) [inline packages/core/lib/src/core_widget_factory.dart:197]: This method should be public, so the svg package can override it to support SVG background.

File: packages/core/test/_.dart [hunk 2]
```diff
@@ -284,11 +284,45 @@ class Explainer {
...
+
+  String _assetImage(AssetImage image) =>
+      '[AssetImage:assetName=${image.assetName}]';
+
+  String _fileImage(FileImage image) => '[FileImage:file=${image.file.path}]';
+
+  String _networkImage(NetworkImage image) => '[NetworkImage:url=${image.url}]';
+
+  String _memoryImage(MemoryImage image) =>
+      '[MemoryImage:size=${image.bytes.length}]';
```
Comment (daohoangson) [inline packages/core/test/_.dart:324]: Flutter's image providers have good `toString` already so maybe we don't need to do this?

**Review by daohoangson — CHANGES_REQUESTED** [review]
Looking great. It just needs a conflict resolve with the `master` branch.

**Comment by anttileppa** [comment]
Merged master and resolved review comments

**Comment by daohoangson** [comment]
Thank you for the resolve. I have merged this PR!

**Pull request merged**